vfs.go

```diff
@@ -396,3 +399,107 @@ func (v *VFS) DirSize(uri string) (uint64, error) {
 
 	return uint64(cfsize), nil
 }
+
+// Close a file. This is flushes the buffered data into the file when the file
+// was opened in write (or append) mode. It is particularly important to be
+// called after S3 writes, as otherwise the writes will not take effect.
+func (v *VFSfh) Close() error {
+
+	ret := C.tiledb_vfs_close(v.context.tiledbContext, v.tiledbVFSfh)
+
+	if ret != C.TILEDB_OK {
+		return fmt.Errorf("Unknown error in VFS.Close: %s", v.context.LastError())
+	}
+
+	v.Free()
+	return nil
+}
+
+// Read part of a file
+func (v *VFSfh) Read(p []byte) (int, error) {
+	nbytes := uint64(len(p))
+
+	// If the size is empty, fetch it
+	if v.size == nil {
+		size, err := v.vfs.FileSize(v.uri)
+		if err != nil {
+			return 0, err
+		}
+		v.size = &size
+	}
+
+	// If the requested read size is beyond the limit, adjust bytes to read
+	if v.offset+nbytes > *v.size {
+		nbytes = *v.size - v.offset
+	}
+
+	if nbytes == 0 {
```

Comment:
Why not check this first and exit early?

Reply:
We can't check this first incase we need to set the nbytes from the size - offset.